Tests/SuperLinq.Test/HasDuplicatesTest.cs

```diff
@@ -0,0 +1,230 @@
+﻿namespace Test;
+
+public class HasDuplicatesTest
+{
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Without_Duplicates_Then_False_Is_Returned()
+	{
+		var stringArray = new[]
+		{
+			"FirstElement",
+			"SecondElement",
+			"ThirdElement"
+		};
+
+		var hasDuplicates = stringArray.HasDuplicates();
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Projection_Without_Duplicates_Then_False_Is_Returned()
+	{
+		var dummyClasses = new DummyClass[]
+		{
+			new("FirstElement"),
+			new("SecondElement"),
+			new("ThirdElement")
+		};
+
+		var hasDuplicates = dummyClasses.HasDuplicates(x => x.ComparableString);
+
+		Assert.False(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_True_Is_Returned()
+	{
+		var stringArray = new[]
+		{
+			"FirstElement",
+			"DUPLICATED_STRING",
+			"DUPLICATED_STRING",
+			"ThirdElement"
+		};
+
+		var hasDuplicates = stringArray.HasDuplicates();
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public void When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_True_Is_Returned()
+	{
+		var dummyClasses = new DummyClass[]
+		{
+			new("FirstElement"),
+			new("DUPLICATED_STRING"),
+			new("DUPLICATED_STRING"),
+			new("ThirdElement")
+		};
+
+		var hasDuplicates = dummyClasses.HasDuplicates(x => x.ComparableString);
+
+		Assert.True(hasDuplicates);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
+	{
+		var source = SuperEnumerable.From(() => "FirstElement",
+			() => "DUPLICATED_STRING",
+			() => "DUPLICATED_STRING",
+			() => throw new TestException());
+
+		var record = await Record.ExceptionAsync(() => Task.FromResult(source.HasDuplicates()));
+
+		Assert.Null(record);
+	}
+
+	[Fact]
+	public async Task When_Asking_For_Duplicates_On_Sequence_Projection_With_Duplicates_Then_It_Does_Not_Iterate_Unnecessary_On_Elements()
+	{
+		var source = SuperEnumerable.From(() => new DummyClass("FirstElement"),
+			() => new DummyClass("DUPLICATED_STRING"),
+			() => new DummyClass("DUPLICATED_STRING"),
+			() => throw new TestException());
+
+		var record = await Record.ExceptionAsync(() => Task.FromResult(source.HasDuplicates(x => x.ComparableString)));
+
+		Assert.Null(record);
```

Comment:
```suggestion
		using var sequence = SeqExceptionAt(4);

		var hasDuplicates = sequence.HasDuplicates(x => x / 2);

		Assert.True(hasDuplicates);
```

Reply:
it does shorten the test, but I find the previous version more readable. 

and reading the proposed version, you kinda need to know about the SeqException implementation, and that 
1/2 = 0 -> round down because of int to 1
2/2 = 1 -> 1
3/2 = 1 -> round down because of int 
4 -> exception here

Reply:
Sure, I just don't like dealing with `.From()` and having to do the `new TestException()` at the end of it. 

If you wanted to add a `TestingSequence.OfWithFailure()`, I'd agree to that.